src/DDTrace/Encoders/SpanEncoder.php

```diff
@@ -15,43 +16,43 @@ final class SpanEncoder
      * @param Span $span
      * @return array
      */
-    public static function encode(Span $span)
+    public static function encode(SpanData $span)
     {
         self::logSpanDetailsIfDebug($span);
 
         $arraySpan = [
-            'trace_id' => (int) $span->getTraceId(),
-            'span_id' => (int) $span->getSpanId(),
-            'name' => $span->getOperationName(),
-            'resource' => $span->getResource(),
-            'service' => $span->getService(),
-            'start' => (int) ($span->getStartTime() . '000'),
-            'error' => $span->hasError() ? 1 : 0,
+            'trace_id' => (int) $span->context->traceId,
+            'span_id' => (int) $span->context->spanId,
+            'name' => $span->operationName,
+            'resource' => $span->resource,
+            'service' => $span->service,
+            'start' => (int) ($span->startTime . '000'),
+            'error' => $span->hasError ? 1 : 0,
         ];
 
-        if ($span->getType() !== null) {
-            $arraySpan['type'] = $span->getType();
+        if ($span->type !== null) {
+            $arraySpan['type'] = $span->type;
         }
 
-        if ($span->isFinished()) {
-            $arraySpan['duration'] = (int) ($span->getDuration() . '000');
+        if ($span->duration != null) { // is span finished ?
```

Comment:
I'd do a strict check here. :)

```suggestion
        if ($span->duration !== null) { // is span finished ?
```